src/crypto/secretstream/secretstream_macros.rs

```diff
@@ -0,0 +1,354 @@
+macro_rules! stream_module (($state_name: ident,
```

Comment:
I’m still not a fan of this being a macro. It seems premature to me, we have no idea if libsodium will add more primitives of this form, and even though it does we don’t know if the function signatures will be the same. I believe this shouldn’t be a macro for now, but might become one in the future if needed.